docs/features/network-security-controls/admin-network-policy.md

```diff
@@ -1,303 +1,1118 @@
+  `direction` can be either `Ingress` or `Egress` and `action` can be either
+  `Pass` or `Allow` or `Deny`.
+* `ovnkube_controller_baseline_admin_network_policies_rules`: The total number of rules
+  across all baseline admin network policies in the cluster grouped by `direction` and `action`.
+  `direction` can be either `Ingress` or `Egress` and `action` can be either
+  `Allow` or `Deny`.
+
+![anp-metrics-1](../../images/anp-metrics-1.png)
+![anp-metrics-2](../../images/anp-metrics-2.png)
+![anp-metrics-3](../../images/anp-metrics-3.png)
+![anp-metrics-4](../../images/anp-metrics-4.png)
+
+## Best Practices and Gotchas!
+
+* Its better to create ANPs at separate priorities so that the precedence
+  is well defined and if you are creating two ANPs at the same priority,
+  then it is the responsibility of the end user to ensure the rules in
+  those ANPs are not overlapping. If it overlaps then its not guaranteed
+  which one will take effect.
+* Unlike ANPs BANP is a singleton object. So there can be only 1 `default`
+  BANP in your cluster. Hence use this policy wisely as a wider default
+  guardrail.
+* API semantics for admin network policies are very different from network
+  policies. There is no implicit deny when you create the policy. You have to
+  explicitly define what you need. Because of this it is not possible to
+  completely isolate the pod from everything using admin network policies for
+  ingress traffic. Only peers you can express for a deny for ingress rule are
+  pods; so atmost only "DENY from all other pods" can be expressed as the best
+  case deny for ingress. In case of egress, one can deny to 0.0.0.0/0 and it will
+  act like a default deny to everything.
+* Unlike network policies, admin network policies are cluster-scoped. So use the
+  empty namespace selector with care because it will select ALL namespaces in your
+  cluster including the infrastructure namespaces like kube-system and kapi-server
+  or openshift-*. Best practice to secure your workloads is to use workload
+  specific labels and then use that as the namespace selector instead.
+* `Pass` is a special case action used to delegate policy decision to the
+  network policy layer. If you don't care about interoperability with existing
+  network policies, then you probably don't need to worry about this feature.
+* Since `0-100` is the supported priority range, it is always better when designing
+  your admin policies to use a middle range like `30-70` by leaving some placeholder
+  priorities before and after. Even in the middle range, be careful to leave gaps so
+  that as your infrastructure requirements evolve over time, you are able to insert
+  new ANPs when needed at the right priority level. If you pack your ANPs, then you
+  might need to recreate all of them to accommodate any changes in the future.
+* Each rule can have upto 100 peers. All those peers are added to the same address-set.
```

Comment:
@igsilya : PTAL this point. Do you have other scale wise recommendations or bugs for ACLs/Address_Sets/PortGroups that we need to translate into OVNK world?

Reply:
Are 'except' blocks allowed in ANP?  If so, I'd suggest discouraging from using these and suggest implementing the same rules with priorities, i.e. high priority rule to allow the IP block and a low priority rule to deny everything, instead of a single rule with an 'except' block.

Reply:
no except blocks are not allowed in ANP, we can either do:

drop if cidr == A,B,C OR
allow if cidr == A,B,C

we cannot do "!="